src/NodaTime/Interval.cs

```diff
@@ -161,11 +161,16 @@ public Instant End
         /// <param name="start">The start of the interval.</param>
         /// <param name="end">The end of the interval.</param>
         [Pure]
-        public void Deconstruct(out Instant start, out Instant end)
+        public void Deconstruct(out Instant? start, out Instant? end)
         {
-            start = Start;
-            end = End;
-        }
+            start = this.start.IsValid
```

Comment:
I'd put each statement on one line - I don't think we need to "explode" it quite this much.